### DISCUSSION TITLE
Helix Sensor Error

### DISCUSSION THREAD

**Discussion by mcondon3** [Security-Onion-Solutions/securityonion#8852] [Unsupported Versions]
Getting an error when trying to install as a Helix sensor from the latest ISO. After it says the installation has failed, I tried to run the so-helix-apikey script and get the error output below. I have also attached the sosetup.log that contains additional errors. 

Any help would be greatly appreciated. Thanks in advance!

[~]$ sudo SecurityOnion/salt/common/tools/sbin/so-helix-apikey 
[sudo] password for trellix: 
Enter your Helix API Key: 
<entered key>
Error response from daemon: No such container: so-logstash
Error: No such container: so-logstash
Restarting Logstash for updated key
[INFO    ] Loading fresh modules for state activity
[ERROR   ] Rendering exception occurred
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 502, in render_jinja_tmpl
    output = template.render(**decoded_context)
  File "/usr/lib/python3.6/site-pa
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/9710249/sosetup.log)
ckages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/usr/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 15, in top-level template code
  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 1155, in make_module
    return TemplateModule(self, self.new_context(vars, shared, locals))
  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 1238, in __init__
    body_stream = list(template.root_render_func(context))
  File "/var/cache/salt/minion/files/base/allowed_states.map.jinja", line 300, in top-level template code
  File "/usr/lib/python3.6/site-packages/jinja2/sandbox.py", line 460, in call
    if not __self.is_safe_callable(__obj):
  File "/usr/lib/python3.6/site-packages/jinja2/sandbox.py", line 360, in is_safe_callable
    getattr(obj, "unsafe_callable", False) or getattr(obj, "alters_data", False)
jinja2.exceptions.UndefinedError: 'None' has no attribute 'append'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 261, in render_tmpl
    output = render_str(tmplstr, context, tmplpath)
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 509, in render_jinja_tmpl
    raise SaltRenderError("Jinja variable {}{}".format(exc, out), buf=tmplstr)
salt.exceptions.SaltRenderError: Jinja variable 'None' has no attribute 'append'
/var/cache/salt/minion/files/base/allowed_states.map.jinja(300):
---
[...]
  {% endif %}

  {% if REDIS and grains.role in ['so-manager', 'so-standalone', 'so-managersearch', 'so-heavynode', 'so-receiver'] %}
    {% do allowed_states.append('redis') %}
  {% endif %}
    <======================
  {% if grains.os == 'CentOS' %}
    {% if not ISAIRGAP %}
      {% do allowed_states.append('yum') %}
    {% endif %}
    {% do allowed_states.append('yum.packages') %}
[...]
---
[CRITICAL] Rendering SLS 'base:logstash' failed: Jinja variable 'None' has no attribute 'append'
/var/cache/salt/minion/files/base/allowed_states.map.jinja(300):
---
[...]
  {% endif %}

  {% if REDIS and grains.role in ['so-manager', 'so-standalone', 'so-managersearch', 'so-heavynode', 'so-receiver'] %}
    {% do allowed_states.append('redis') %}
  {% endif %}
    <======================
  {% if grains.os == 'CentOS' %}
    {% if not ISAIRGAP %}
      {% do allowed_states.append('yum') %}
    {% endif %}
    {% do allowed_states.append('yum.packages') %}
[...]
---
local:
    Data failed to compile:
----------
    Rendering SLS 'base:logstash' failed: Jinja variable 'None' has no attribute 'append'
/var/cache/salt/minion/files/base/allowed_states.map.jinja(300):
---
[...]
  {% endif %}

  {% if REDIS and grains.role in ['so-manager', 'so-standalone', 'so-managersearch', 'so-heavynode', 'so-receiver'] %}
    {% do allowed_states.append('redis') %}
  {% endif %}
    <======================
  {% if grains.os == 'CentOS' %}
    {% if not ISAIRGAP %}
      {% do allowed_states.append('yum') %}
    {% endif %}
    {% do allowed_states.append('yum.packages') %}
[...]
---

**Comment by dougburks** [comment]
From your `sosetup.log`:
> Security Onion Version: 2.3.160

Have you tried the latest 2.3.180 version?
https://blog.securityonion.net/2022/10/security-onion-23180-now-available.html
